### PR TITLE
refactor: refactor the logs of lvmcluster controller

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

**Review by leelavg — COMMENTED** [review]
- general observation, do we need to log error/info in both call and execution site (ex: `getRunningPodImage`)?
- it seems PR increased number of info level logs, (ex: `openshiftsccs`)
- when we debug issues, we first jump into error site and go up the stack and logging similar errors may not be helpful always (this is only my opinion)

File: controllers/lvmcluster_controller.go [hunk 1]
```diff
@@ -183,7 +186,8 @@ func (r *LVMClusterReconciler) reconcile(ctx context.Context, instance *lvmv1alp
 	for _, unit := range resourceCreationList {
 		err := unit.ensureCreated(r, ctx, instance)
 		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed reconciling: %s %w", unit.getName(), err)
+			r.Log.Error(err, "failed reconciling", "Resource", unit.getName())
```
Comment (nbalacha) [inline controllers/lvmcluster_controller.go:189]: This is actually changing the error being returned, right?

File: controllers/lvmcluster_controller.go [hunk 2]
```diff
@@ -309,12 +322,15 @@ func (r *LVMClusterReconciler) getRunningPodImage(ctx context.Context) error {
...
-			return fmt.Errorf("failed to get pod name env variable")
+			err := fmt.Errorf("failed to get pod name env variable")
+			r.Log.Error(err, "POD_NAME env variable is not set.")
+			return err
 		}
 
 		pod := &corev1.Pod{}
 		if err := r.Get(ctx, types.NamespacedName{Name: podName, Namespace: r.Namespace}, pod); err != nil {
-			return fmt.Errorf("failed to get pod %s in namespace %s", podName, r.Namespace)
+			r.Log.Error(err, "Failed to get pod.")
```
Comment (nbalacha) [inline controllers/lvmcluster_controller.go:332]: Will this include information about the pod that was requested?

**Review by nbalacha — COMMENTED** [review]
Can you post the before and after logs for the operator so we can see what the changes look like?

File: controllers/lvmcluster_controller.go (hunk 1 above)
Comment (iamniting) [reply controllers/lvmcluster_controller.go:189]: yes I know, We should return an original error IMO as that can be used to determine the error in the caller's place.

File: controllers/lvmcluster_controller.go (hunk 2 above)
Comment (iamniting) [reply controllers/lvmcluster_controller.go:332]: Yes error will have it, This is the error which gets generated `pods "test" not found` when I ran the below code.

```
pod := &corev1.Pod{}                                    
err = fakeReconciler.Client.Get(context.TODO(), types.NamespacedName{Name: "test", Namespace: "test"}, pod)                                                       
fmt.Println(err)
```

**Comment by iamniting** [comment]
> Can you post the before and after logs for the operator so we can see what the changes look like?

[after-lvm.log](https://github.com/red-hat-storage/lvm-operator/files/8386864/after-lvm.log)

**Comment by nbalacha** [comment]
Can the following be done?
1. Remove` "reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster"` from the messages. I think they are unnecessary.
2. Rework the messages to remove duplicate fields. For instance, the req name and namespace already have the information captured in name and namespace in the messages below:
```
{"level":"info","ts":1648710461.5450764,"logger":"controller.lvmcluster.lvmcluster-controller","msg":"openshiftSCC found, setting cluster type to openshift.","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster","name":"lvmcluster-sample","namespace":"lvm-operator-system","Request.Namespace":"lvm-operator-system","Request.Name":"lvmcluster-sample"}
{"level":"info","ts":1648710461.6462138,"logger":"controller.lvmcluster.lvmcluster-controller","msg":"Finalizer not found for LvmCluster. Adding finalizer.","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster","name":"lvmcluster-sample","namespace":"lvm-operator-system","Request.Namespace":"lvm-operator-system","Request.Name":"lvmcluster-sample"}
```
3. Uniform capitalization for messages. The older ones do not start with a capital letter. 
4. uniform capitalization for field names. Older ones use all smallcase letters for the most part.
5. Reconsider some of the existing log fields. For instance:
`{"level":"info","ts":1648710462.1055522,"logger":"controller.lvmcluster.lvmcluster-controller","msg":"already exists","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster","name":"lvmcluster-sample","namespace":"lvm-operator-system","Request.Namespace":"lvm-operator-system","Request.Name":"lvmcluster-sample","SecurityContextConstraint":"odf-lvm-vgmanager"}`
may be better as 
{"level":"info","ts":1648710462.1055522,"logger":"controller.lvmcluster.lvmcluster-controller","msg":"already exists","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster","name":"lvmcluster-sample","namespace":"lvm-operator-system","Request.Namespace":"lvm-operator-system","Request.Name":"lvmcluster-sample", "type","SecurityContextConstraint", "name", "odf-lvm-vgmanager"}
or
{"level":"info","ts":1648710462.1055522,"logger":"controller.lvmcluster.lvmcluster-controller","msg":"SCC odf-lvm-vgmanager already exists","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster","name":"lvmcluster-sample","namespace":"lvm-operator-system","Request.Namespace":"lvm-operator-system","Request.Name":"lvmcluster-sample"}

**Comment by iamniting** [comment]
> Can the following be done?
> 
> 1. Remove` "reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster"` from the messages. I think they are unnecessary.
> 2. Rework the messages to remove duplicate fields. For instance, the req name and namespace already have the information captured in name and namespace in the messages below:
> 
> ```
> {"level":"info","ts":1648710461.5450764,"logger":"controller.lvmcluster.lvmcluster-controller","msg":"openshiftSCC found, setting cluster type to openshift.","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster","name":"lvmcluster-sample","namespace":"lvm-operator-system","Request.Namespace":"lvm-operator-system","Request.Name":"lvmcluster-sample"}
> {"level":"info","ts":1648710461.6462138,"logger":"controller.lvmcluster.lvmcluster-controller","msg":"Finalizer not found for LvmCluster. Adding finalizer.","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster","name":"lvmcluster-sample","namespace":"lvm-operator-system","Request.Namespace":"lvm-operator-system","Request.Name":"lvmcluster-sample"}
> ```
> 
> 3. Uniform capitalization for messages. The older ones do not start with a capital letter.
> 4. uniform capitalization for field names. Older ones use all smallcase letters for the most part.
> 5. Reconsider some of the existing log fields. For instance:
>    `{"level":"info","ts":1648710462.1055522,"logger":"controller.lvmcluster.lvmcluster-controller","msg":"already exists","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster","name":"lvmcluster-sample","namespace":"lvm-operator-system","Request.Namespace":"lvm-operator-system","Request.Name":"lvmcluster-sample","SecurityContextConstraint":"odf-lvm-vgmanager"}`
>    may be better as
>    {"level":"info","ts":1648710462.1055522,"logger":"controller.lvmcluster.lvmcluster-controller","msg":"already exists","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster","name":"lvmcluster-sample","namespace":"lvm-operator-system","Request.Namespace":"lvm-operator-system","Request.Name":"lvmcluster-sample", "type","SecurityContextConstraint", "name", "odf-lvm-vgmanager"}
>    or
>    {"level":"info","ts":1648710462.1055522,"logger":"controller.lvmcluster.lvmcluster-controller","msg":"SCC odf-lvm-vgmanager already exists","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMCluster","name":"lvmcluster-sample","namespace":"lvm-operator-system","Request.Namespace":"lvm-operator-system","Request.Name":"lvmcluster-sample"}

made most of the changes, pls take a look.

File: controllers/lvmcluster_controller.go [hunk 3]
```diff
@@ -309,12 +322,15 @@ func (r *LVMClusterReconciler) getRunningPodImage(ctx context.Context) error {
...
-			return fmt.Errorf("failed to get pod name env variable")
+			err := fmt.Errorf("failed to get pod name env variable")
+			r.Log.Error(err, "POD_NAME env variable is not set")
+			return err
 		}
 
 		pod := &corev1.Pod{}
 		if err := r.Get(ctx, types.NamespacedName{Name: podName, Namespace: r.Namespace}, pod); err != nil {
-			return fmt.Errorf("failed to get pod %s in namespace %s", podName, r.Namespace)
+			r.Log.Error(err, "failed to get pod")
```
Comment (leelavg) [inline controllers/lvmcluster_controller.go:332]: - better log at-least pod name

Comment (nbalacha) [reply controllers/lvmcluster_controller.go:332]: agree

File: controllers/scc.go [hunk 4]
```diff
@@ -52,14 +52,16 @@ func (c openshiftSccs) ensureCreated(r *LVMClusterReconciler, ctx context.Contex
 			r.Log.Info("creating SecurityContextConstraint", "SecurityContextConstraint", scc.Name)
 			_, err := r.SecurityClient.SecurityContextConstraints().Create(ctx, scc, metav1.CreateOptions{})
 			if err != nil {
-				return fmt.Errorf("failed to create SCC %q: %v", scc.Name, err)
```
Comment (nbalacha) [inline controllers/scc.go:55]: Please include the scc name in the error and the log

Comment (iamniting) [reply controllers/scc.go:55]: err will already have it, But I have added it again in the logs.

File: controllers/lvmcluster_controller.go [hunk 5]
```diff
@@ -324,10 +340,11 @@ func (r *LVMClusterReconciler) getRunningPodImage(ctx context.Context) error {
 			}
 		}
 
-		return fmt.Errorf("failed to get container image for %s in pod %s", LVMOperatorContainerName, podName)
+		err := fmt.Errorf("failed to get container image for %s in pod %s", LVMOperatorContainerName, podName)
+		r.Log.Error(err, "container image not found")
```
[the remark this answers is not in the document]
Comment (iamniting) [reply controllers/lvmcluster_controller.go:344]: We already have if you see above created err.

File: controllers/lvmcluster_controller.go (hunk 3 above)
Comment (iamniting) [reply controllers/lvmcluster_controller.go:332]: err will have it, but still i added in the logs again.

File: controllers/lvmcluster_controller.go [hunk 6]
```diff
@@ -183,7 +186,8 @@ func (r *LVMClusterReconciler) reconcile(ctx context.Context, instance *lvmv1alp
 	for _, unit := range resourceCreationList {
 		err := unit.ensureCreated(r, ctx, instance)
 		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed reconciling: %s %w", unit.getName(), err)
+			r.Log.Error(err, "failed reconciling", "resource", unit.getName())
```
[the remark this answers is not in the document]
Comment (iamniting) [reply controllers/lvmcluster_controller.go:189]: done

**Review by nbalacha — APPROVED** [review]

**Review by leelavg — APPROVED** [review]
/lgtm

**Pull request merged**